tests/testthat/test-training.R

```diff
@@ -32,3 +32,122 @@ test_that("plot.trainSLOPE works as expected", {
   p <- plot(fit)
   expect_s3_class(p, "trellis")
 })
+
+
+test_that("trainSLOPE works properly for binomial family", {
+  set.seed(42)
+  xy <- SLOPE:::randomProblem(200, p=100, q=0.5, response="binomial")
+  x <- xy$x
+  y <- xy$y
+
+  fit <- trainSLOPE(x, y, q = c(0.1, 0.2), number = 2, family = "binomial")
+
+  expect_equal(fit$measure$measure, c("mse", "mae", "deviance", "misclass", "auc"))
+  expect_equal(fit$measure$label, c("Mean Squared Error", "Mean Absolute Error", "Binomial Deviance",
+                                    "Misclassification Rate", "AUC"))
+
+  expect_equal(fit$optima, structure(list(q = c(0.1, 0.2, 0.2, 0.1, 0.2),
+                                          alpha = c(0.0407419692328029, 0.0095161892230067, 7.46792437250563e-05, 0.00586052810915448, 0.0095161892230067),
+                                          measure = c("auc", "deviance", "mae", "misclass", "mse"),
+                                          mean = c(0.660558191808191, 0.980242385828498, 0.505232740237056, 0.235, 0.323146578355248),
+                                          se = c(0.0427697302697303, 0.0895079147155515, 0.011858410680709, 0.055, 0.0303254835186373),
+                                          lo = c(0.11711724249003, -0.157063504055364, 0.354557346282327, -0.463841260489608, -0.0621752239560481),
+                                          hi = c(1.20399914112635, 2.11754827571236, 0.655908134191785, 0.933841260489608, 0.708468380666544)),
+                                     row.names = c(NA, -5L),
+                                     class = "data.frame"),
+               tolerance = 0.0001)
+})
```

Comment:
Can you please reformat this? Lines should be no wider than 80 characters.

Reply:
Fixed